frontend/src/types.ts

```diff
@@ -12,6 +12,7 @@ export interface MessageDocument {
 export interface Message {
   id: string;
   type: string;
+  role?: string; // for chat_retrieval bot
```

Comment:
I couldn't decide what to do about this...

Calling `POST /runs/stream` requires the `role` field in the message. Otherwise, the request body will not be serialized appropriately in the backend and an error is raised `langgraph.graph.message.add_messages` (state reducer function).

It seems more appropriate for the client (frontend) to handle this instead of the API (backend), but I'm not sure.